src/api/java/com/minecolonies/api/colony/requestsystem/factory/IFactoryController.java

```diff
@@ -0,0 +1,132 @@
+package com.minecolonies.api.colony.requestsystem.factory;
+
+import net.minecraft.nbt.NBTTagCompound;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface used to describe classes that function as Factory controllers.
+ */
+public interface IFactoryController
+{
+
+    /**
+     * Method used to get a factory for a given input class name.
+     *
+     * @param className The class name of the input type of the requested factory.
+     * @param <Input>   The type of input for the requested factory.
+     * @return The factory that can handle the given input class.
+     *
+     * @throws IllegalArgumentException is thrown when the given input name is unknown to this Factory Controller.
+     */
+    default <Input> IFactory<Input, ?> getFactoryForInput(@NotNull String className) throws IllegalArgumentException
+    {
+        //Simple default implementation grabs the class from a given name and casts it to the proper type.
+        //Any exceptions thrown before actual request is made gets wrapped.
+        try
+        {
+            return getFactoryForInput((Class<? extends Input>) Class.forName(className));
+        }
+        catch (IllegalArgumentException ex)
```

Comment:
Nope i actually forcefully want to rethrow here. I singled out the IllegalArgumentExcetption as it is part of the method declaration.